### ISSUE TITLE
bug: conform with gofumpt doesn't seem to be correct

### ISSUE THREAD

**Issue by nnathan** [stevearc/conform.nvim#560]
### Neovim version (nvim -v)

v0.11.0-dev

### Operating system/version

Ubuntu 24.04

### Read debugging tips

- [X] I have read through the [debugging tips](https://github.com/stevearc/conform.nvim/blob/master/doc/debugging.md).

### Add the debug logs

- [X] I have set `log_level = vim.log.levels.DEBUG` and pasted the log contents below.

### Log file

```
2024-10-27 12:51:59[INFO] Run gofumpt on /home/naveen/code/mocknet/pkg/servicemocks/dockerhub_index.go

2024-10-27 12:51:59[TRACE] Input lines: { "package servicemocks", "", "import (", '\t"context"', '\t"encoding/json"', '\t"mocknet/pkg/datastore"', '\t"mocknet/pkg/mirrorstore/mirrorstorefs"', '\t"net/http"', '\t"path/filepath"', '\t"slices"', '\t"strconv"', '\t"strings"', "", '\t"github.com/gorilla/mux"', '\t"go.uber.org/zap"', ")", "", "type repositoryMetadata struct {", "\tName        string", '\tDescription string `json:"description"`', '\tStarCount   int    `json:"star_count"`', '\tOfficial    bool   `json:"is_official"`', "}",

.
.
.

2024-10-27 12:51:59[TRACE] Output lines: { "package servicemocks", "", "import (", '\t"context"', '\t"encoding/json"', '\t"mocknet/pkg/datastore"', '\t"mocknet/pkg/mirro
rstore/mirrorstorefs"', '\t"net/http"', '\t"path/filepath"', '\t"slices"', '\t"strconv"', '\t"strings"', "", '\t"github.com/gorilla/mux"', '\t"go.uber.org/zap"', ")", "", "type repositoryMetadata struct {", "\tName        string", '\tDescription string `json:"description"`', '\tStarCount   int    `json:"star_count"`', '\tOfficial    bool   `json:"is_official"`', "}",

.
.
.

2024-10-27 12:52:52[TRACE] Diff indices {}
2024-10-27 12:52:52[TRACE] Applying text edits: {}
2024-10-27 12:52:52[TRACE] Done formatting /home/naveen/code/mocknet/pkg/servicemocks/dockerhub_index.go
```

### Describe the bug

When running `gofumpt` via conform in neovim it doesn't appear to change the import lines.

However running the same gofumpt binary on the host shows that gofumpt would make changes:

```
$ /home/naveen/.local/share/nvim/mason/bin/gofumpt -d dockerhub_index.go
diff dockerhub_index.go.orig dockerhub_index.go
--- dockerhub_index.go.orig
+++ dockerhub_index.go
@@ -3,8 +3,6 @@
 import (
        "context"
        "encoding/json"
-       "mocknet/pkg/datastore"
-       "mocknet/pkg/mirrorstore/mirrorstorefs"
        "net/http"
        "path/filepath"
        "slices"
@@ -11,6 +9,9 @@
        "strconv"
        "strings"

+       "mocknet/pkg/datastore"
+       "mocknet/pkg/mirrorstore/mirrorstorefs"
+
        "github.com/gorilla/mux"
        "go.uber.org/zap"
 )
```

Version info:
```
$ /home/naveen/.local/share/nvim/mason/bin/gofumpt -version
v0.7.0 (go1.22.2)
```

I said this is blocking because gofumpt wants to incorrectly format things, and so I can't use the plugin with gofumpt.

### What is the severity of this bug?

blocking (cannot use plugin)

### Steps To Reproduce

Have a file with the following:

```
package servicemocks

import (
        "context"
        "encoding/json"
        "mocknet/pkg/datastore"
        "mocknet/pkg/mirrorstore/mirrorstorefs"
        "net/http"
        "path/filepath"
        "slices"
        "strconv"
        "strings"

        "github.com/gorilla/mux"
        "go.uber.org/zap"
)
```

Try to have conform format with gofumpt.

### Expected Behavior

The file should save like this:

```
package servicemocks

import (
        "context"
        "encoding/json"
        "net/http"
        "path/filepath"
        "slices"
        "strconv"
        "strings"

        "mocknet/pkg/datastore"
        "mocknet/pkg/mirrorstore/mirrorstorefs"

        "github.com/gorilla/mux"
        "go.uber.org/zap"
)
```

### Minimal example file

```
package servicemocks

import (
        "context"
        "encoding/json"
        "mocknet/pkg/datastore"
        "mocknet/pkg/mirrorstore/mirrorstorefs"
        "net/http"
        "path/filepath"
        "slices"
        "strconv"
        "strings"

        "github.com/gorilla/mux"
        "go.uber.org/zap"
)
```

### Minimal init.lua

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  {
    "stevearc/conform.nvim",
    config = function()
      require("conform").setup({
        log_level = vim.log.levels.DEBUG,
        -- add your config here
      })
    end,
  },
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by nnathan** [comment]
Looks like the bug is with gofumpt.

```
$ cat foo.go
package servicemocks

import (
        "context"
        "encoding/json"
        "net/http"
        "path/filepath"
        "slices"
        "strconv"
        "strings"

        "mocknet/pkg/datastore"
        "mocknet/pkg/mirrorstore/mirrorstorefs"

        "github.com/gorilla/mux"
        "go.uber.org/zap"
)
```

```
$ gofumpt -d foo.go
$ echo $?
0
```

```
$ cat foo.go | gofumpt
package servicemocks

import (
        "context"
        "encoding/json"
        "mocknet/pkg/datastore"
        "mocknet/pkg/mirrorstore/mirrorstorefs"
        "net/http"
        "path/filepath"
        "slices"
        "strconv"
        "strings"

        "github.com/gorilla/mux"
        "go.uber.org/zap"
)
```

**Comment by nnathan** [comment]
Found a similar issue here: https://github.com/stevearc/conform.nvim/issues/387 which links back to the gofumpt issue tracker.

The problem is my go module didn't have a dot in it and so it was deemed apart of the stdlib and therefore being treated as such. I added a fake `.tld` to the end of the module and changed all the imports and now gofumpt works the same via stdin (the way conform does it) and via file with knowledge of the module.
